pkg/resources/masking_policy_acceptance_test.go

```diff
@@ -216,3 +369,212 @@ func TestAcc_MaskingPolicy_migrateFromVersion_0_94_1(t *testing.T) {
+				ConfigVariables: tfconfig.ConfigVariablesFromModel(t, policyModel),
+				ExpectError:     regexp.MustCompile(`invalid data type: invalid-type`),
+			},
+		},
+	})
+}
+
+func TestAcc_MaskingPolicy_DataTypeAliases(t *testing.T) {
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+	resourceName := "snowflake_masking_policy.test"
+	body := "case when current_role() in ('ANALYST') then 'ok' else '***' end"
+	policyModel := model.MaskingPolicy("test", []sdk.TableColumnSignature{
+		{
+			Name: "a",
+			Type: "TEXT",
+		},
+	}, body, id.DatabaseName(), id.Name(), string(sdk.DataTypeVARCHAR), id.SchemaName())
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		Steps: []resource.TestStep{
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_MaskingPolicy/basic"),
+				ConfigVariables: tfconfig.ConfigVariablesFromModel(t, policyModel),
+				Check: assert.AssertThat(t, resourceassert.MaskingPolicyResource(t, resourceName).
+					HasNameString(id.Name()).
+					HasReturnDataTypeString(string(sdk.DataTypeVARCHAR)).
+					HasArguments([]sdk.TableColumnSignature{
+						{
+							Name: "a",
+							Type: sdk.DataTypeVARCHAR,
+						},
+					}),
+				),
+			},
+		},
+	})
+}
+
+func TestAcc_MaskingPolicy_migrateFromVersion_0_95_0(t *testing.T) {
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+	resourceName := "snowflake_masking_policy.test"
+	comment := "Terraform test resource"
+	body := "case when current_role() in ('ANALYST') then 'true' else 'false' end"
+	policyModel := model.MaskingPolicy("test", []sdk.TableColumnSignature{
+		{
+			Name: "A",
+			Type: sdk.DataTypeVARCHAR,
+		},
+		{
+			Name: "b",
+			Type: sdk.DataTypeVARCHAR,
+		},
+	}, body, id.DatabaseName(), id.Name(), string(sdk.DataTypeVARCHAR), id.SchemaName()).WithComment(comment).WithExemptOtherPolicies(r.BooleanTrue)
+
+	resource.Test(t, resource.TestCase{
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		PreCheck: func() { acc.TestAccPreCheck(t) },
+		Steps: []resource.TestStep{
+			{
+				ExternalProviders: map[string]resource.ExternalProvider{
+					"snowflake": {
+						VersionConstraint: "=0.95.0",
+						Source:            "Snowflake-Labs/snowflake",
+					},
+				},
+				Config: maskingPolicy_v0_95_0(id, body, comment),
+				Check: assert.AssertThat(t, resourceassert.MaskingPolicyResource(t, resourceName).
+					HasNameString(id.Name()).
+					HasDatabaseString(id.DatabaseName()).
+					HasSchemaString(id.SchemaName()).
+					HasCommentString(comment).
+					HasFullyQualifiedNameString(id.FullyQualifiedName()),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "masking_expression", body)),
```

Comment:
add one assert on "id" field to make sure it changed

Reply:
Done.